### PR TITLE
Add support for Physical Servers refresh

### PR DESCRIPTION
**Pull request by MaysaMacedo**
This PR enables:
- Refresh of one or more Physical Servers.

Related to [#2547](https://github.com/ManageIQ/manageiq-ui-classic/pull/2547)

### CODE REVIEW

**Comment by MaysaMacedo** [comment]
@agrare Could you take a look on this PR?

**Comment by MaysaMacedo** [comment]
@miq-bot add_lab gaprindashvili/yes, enhancement

**Comment by MaysaMacedo** [comment]
@miq-bot add_label gaprindashvili/yes,enhancement

File: app/controllers/api/physical_servers_controller.rb [hunk 1]
```diff
@@ -40,6 +40,24 @@ def restart_mgmt_controller_resource(type, id, _data)
       change_resource_state(:restart_mgmt_controller, type, id)
     end
 
+    def refresh_physical_server(physical_server)
```
Comment (abellotti) [inline app/controllers/api/physical_servers_controller.rb]: maybe move this to private section

Comment (MaysaMacedo) [reply app/controllers/api/physical_servers_controller.rb]: @abellotti @jntullo Thanks for your feedback, just fixed it.

**Comment by MaysaMacedo** [comment]
@abellotti @jntullo Thanks for your feedback, I've updated it.

**Comment by AndreyMenezes** [comment]
@abellotti @jntullo Could anyone review this PR?

File: app/controllers/api/physical_servers_controller.rb [hunk 2]
```diff
@@ -40,6 +40,16 @@ def restart_mgmt_controller_resource(type, id, _data)
       change_resource_state(:restart_mgmt_controller, type, id)
     end
 
+    def refresh_resource(type, id = nil, _data = nil)
+      raise BadRequestError, "Must specify an id for refreshing a #{type} resource" unless id
```
Comment (imtayadeway) [inline app/controllers/api/physical_servers_controller.rb]: Can you add a test for this condition?

File: app/controllers/api/physical_servers_controller.rb [hunk 3]
```diff
@@ -40,6 +40,16 @@ def restart_mgmt_controller_resource(type, id, _data)
       change_resource_state(:restart_mgmt_controller, type, id)
     end
 
+    def refresh_resource(type, id = nil, _data = nil)
```
Comment (imtayadeway) [inline app/controllers/api/physical_servers_controller.rb]: Is there a reason for the default `id = nil` here? Does anything call this without the second argument?

File: spec/requests/physical_servers_spec.rb [hunk 4]
```diff
@@ -304,4 +304,66 @@
       end
     end
   end
+
+  describe "Physical Server refresh action" do
+    context "with an invalid id" do
+      it "fails to refresh a physical server" do
```
Comment (imtayadeway) [inline spec/requests/physical_servers_spec.rb]: Nit, but `it "fails..."` suggests that something is wrong....possibly "it responds with 404 Not Found" or similar?

File: spec/requests/physical_servers_spec.rb [hunk 5]
```diff
@@ -304,4 +304,66 @@
...
+        post(api_physical_server_url(nil, 999_999), :params => gen_request(:refresh))
+
+        expect(response).to have_http_status(:not_found)
+      end
+    end
+
+    context "without an appropriate role" do
+      it "fails to refresh a physical server" do
+        ps = FactoryGirl.create(:physical_server)
+
```
Comment (jntullo) [inline spec/requests/physical_servers_spec.rb]: nit - can you remove this empty line? setting up the authorization and creating the factory are both part of the setup, they can be together. Same with similar instances below.

File: spec/requests/physical_servers_spec.rb [hunk 6]
```diff
@@ -304,4 +304,66 @@
...
+        api_basic_authorize(action_identifier(:physical_servers, :refresh, :resource_actions, :post))
+
+        post(api_physical_server_url(nil, 999_999), :params => gen_request(:refresh))
+
+        expect(response).to have_http_status(:not_found)
+      end
+    end
+
+    context "without an appropriate role" do
+      it "fails to refresh a physical server" do
```
Comment (imtayadeway) [inline spec/requests/physical_servers_spec.rb]: Similarly....avoid variations of `it "fails...."`

File: app/controllers/api/physical_servers_controller.rb (hunk 3 above)
Comment (MaysaMacedo) [reply app/controllers/api/physical_servers_controller.rb]: You are right. No need for this here.

**Comment by MaysaMacedo** [comment]
@imtayadeway  @jntullo Thanks! Just updated.

**Comment by abellotti** [comment]
LGTM!! Thanks @MaysaMacedo for the API Enhancement!! 👍

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 8f63787c368bf3eec14575bc2b24a1be3c66426f
Author: Alberto Bellotti <abellotti@users.noreply.github.com>
Date:   Tue Nov 21 10:55:52 2017 -0500

    Merge pull request #189 from MaysaMacedo/physical_server_refresh
    
    Add support for Physical Servers refresh
    (cherry picked from commit 2311c73ec9b8569d1e4c6c5588201e31f775b6dd)
```

**Pull request merged**